### PR TITLE
/status page only on the production network

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Only allow /status page to be accessed from the production network
## Impacts

Impacts the access to the status page from the different networks
## NEWS file entries

New Features
++++++++++++
Add the ability to restrict the /status page to the production network
## Delete branch after merge

NO

### CODE REVIEW

File: conf/documentation.conf [hunk 1]
```diff
@@ -940,6 +940,14 @@ description=<<EOT
 Force the captive portal to use HTTPS when redirecting captured clients.
 EOT
 
+[captive_portal.status_only_on_production]
+type=toggle
+options=enabled|disabled
+description=<<EOT
+If status only onproduction is enabled, the /status page will only be available on the
```
Comment (extrafu) [inline conf/documentation.conf]: ugh what? I can't grok this.

File: html/pfappserver/lib/pfappserver/I18N/i_default.po [hunk 2]
```diff
@@ -1808,6 +1808,12 @@ msgid ""
   "forward to the captive portal."
 msgstr ""
 
+# conf/documentation.conf (captive_portal.status_only_on_production)
+msgid ""
+  "If status only onproduction is enabled, the /status page will only be available on the"
```
Comment (extrafu) [inline html/pfappserver/lib/pfappserver/I18N/i_default.po]: same

File: lib/pf/services/manager/httpd.pm [hunk 3]
```diff
@@ -89,20 +89,26 @@ sub generateConfig {
     $tags{'proxies'} = join( "\n", @proxies );
 
     # Guest related URLs allowed through Apache ACL's
-    $tags{'allowed_from_all_urls'} = "|$WEB::URL_STATUS";
+    my $status_only_on_production = isenabled($Config{captive_portal}{status_only_on_production});
+    print "$status_only_on_production\n";
```
Comment (fdurand) [inline lib/pf/services/manager/httpd.pm]: print ?

File: lib/pf/services/manager/httpd.pm [hunk 4]
```diff
@@ -89,20 +89,26 @@ sub generateConfig {
...
     }
     # /activate/email allowed if sponsor or email mode enabled
     my $email_enabled = $guest_self_registration{$SELFREG_MODE_EMAIL};
     my $sponsor_enabled = $guest_self_registration{$SELFREG_MODE_SPONSOR};
     if ($guest_regist_allowed && ($email_enabled || $sponsor_enabled)) {
         # | is for a regexp "or" as this is pulled from a 'Location ~' statement
-        $tags{'allowed_from_all_urls'} .= "|$WEB::URL_EMAIL_ACTIVATION|$WEB::CGI_EMAIL_ACTIVATION";
+        $allowed_from_all_urls .= "|$WEB::URL_EMAIL_ACTIVATION|$WEB::CGI_EMAIL_ACTIVATION";
     }
+    $tags{'allowed_from_all_urls'} = $allowed_from_all_urls;
```
Comment (fdurand) [inline lib/pf/services/manager/httpd.pm:110]: We will probably have to check that later but it looks that we replicate apache configuration in httpd.portal and in captive-portal-common.conf

**Comment by extrafu** [comment]
Bump?

**Comment by jrouzierinverse** [comment]
Please review

**Comment by jrouzierinverse** [comment]
Please review

**Pull request merged**